src/data/blob.rs

```diff
@@ -0,0 +1,337 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under the MIT license <LICENSE-MIT
+// https://opensource.org/licenses/MIT> or the Modified BSD license <LICENSE-BSD
+// https://opensource.org/licenses/BSD-3-Clause>, at your option. This file may not be copied,
+// modified, or distributed except according to those terms. Please review the Licences for the
+// specific language governing permissions and limitations relating to use of the SAFE Network
+// Software.
+
+use crate::{utils, Error, PublicKey, XorName};
+use bincode::serialized_size;
+use multibase::Decodable;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    fmt::{self, Debug, Formatter},
+    u64,
+};
+use tiny_keccak;
+
+/// Maximum allowed size for a serialised blob (ID) to grow to
+pub const MAX_BLOB_SIZE_IN_BYTES: u64 = 1024 * 1024 + 10 * 1024;
+
+/// PrivateBlob: an immutable chunk of data which can be deleted. Can only be fetched
+/// by the listed owner.
+#[derive(Hash, Eq, PartialEq, PartialOrd, Ord, Clone)]
+pub struct PrivateBlob {
+    /// Address.
+    address: Address,
+    /// Contained data.
+    value: Vec<u8>,
+    /// Contains a set of owners of this data. DataHandlers enforce that a DELETE or OWNED-GET type
+    /// of request is coming from the ClientHandler of the owners.
+    owner: PublicKey,
```

Comment:
You mean for PublicKey? I don't know, nothing changed there from previous implementation.

Reply:
well, the comment says `Contains a set of owners of this data`
maybe this shall be BlsKey ?

Reply:
I think the end goal was this, to have it as `BlsKey`. I did ask around in slack a few weeks back for this, unfortunately I couldn't get it cleared out at the time. 
But I see no apparent reason why not doing it, I have always assumed `BlsKeys` will be the way we have multiple ownership of any and everything in the network.

Reply:
Saw now that the question of PublicKey is moot, since it already is defined like this:

```
pub enum PublicKey {
    Ed25519(ed25519_dalek::PublicKey),
    Bls(threshold_crypto::PublicKey),
    BlsShare(threshold_crypto::PublicKeyShare),
}
```

But docs could be improved as you say.

Reply:
Ah, that will be great.